sql/sys_vars.cc

```diff
@@ -2452,6 +2452,41 @@ Sys_gtid_ignore_duplicates(
        DEFAULT(FALSE), NO_MUTEX_GUARD,
        NOT_IN_BINLOG, ON_CHECK(check_gtid_ignore_duplicates),
        ON_UPDATE(fix_gtid_ignore_duplicates));
+
+static bool
+check_slave_max_statement_time(sys_var *self, THD *thd, set_var *var)
+{
+  return give_error_if_slave_running(0);
```

Comment:
why is that?

Reply:
It's a standard policy pattern that is followed. A number of slave side features are allowed activation
only when the slave is stopped.
It's not a MUST in this case to my view as well, but feels consistent to me.

Reply:
there's no "standard policy pattern". There are slave settings that apply **at start**, like number of parallel workers or `gtid_slave_pos` — they obviously cannot be changed after the slave has already started. `slave_max_statement_time` on the contrary applies to the next event, it can be freely changed anytime with an immediate effect.

Reply:
You're right, some of the slave global variables just can not be changed in run time.
And I should've said that the pattern that I mean (let me leave out its description for another time) is not followed long time ago.
That also makes me fine to relax this specific one.